src/testFixtures/kotlin/com/exactpro/th2/common/TestUtils.kt

```diff
@@ -53,26 +60,81 @@ fun assertEqualBatches(expected: MessageBatch, actual: MessageBatch, lazyMessage
+fun assertEqualGroups(expected: MessageGroupOrBuilder, actual: MessageGroupOrBuilder, lazyMessage: () -> String? = {null}) {
+    Assertions.assertEquals(expected.messagesCount, actual.messagesCount) {"wrong count of messages in group: \n${TextFormat.shortDebugString(actual)}"}
+    expected.messagesList.forEachIndexed { i, message ->
+        try {
+            assertEqualMessages(message, actual.messagesList[i], lazyMessage)
+        } catch (e: AssertionFailedError) {
+            throw AssertionFailedError(
+                "Error in message from group with index '$i'.\n${e.message}",
+                e.expected,
+                e.actual,
+                e.cause
+            )
+        }
+    }
+}
+
+fun assertEqualGroupBatches(expected: MessageGroupBatchOrBuilder, actual: MessageGroupBatchOrBuilder, lazyMessage: () -> String? = {null}) {
+    Assertions.assertEquals(expected.groupsCount, actual.groupsCount) {"wrong count of groups in batch: \n${TextFormat.shortDebugString(actual)}"}
+    expected.groupsList.forEachIndexed { i, message ->
+        assertEqualGroups(message, actual.getGroups(i), lazyMessage)
+    }
+}
+
+fun assertEqualMessages(expected: AnyMessageOrBuilder, actual: AnyMessageOrBuilder, lazyMessage: () -> String? = {null}) {
+    Assertions.assertEquals(expected.kindCase, actual.kindCase) { "Wrong message kind" }
     val ts = Timestamp.getDefaultInstance()
-    val assertExpected = expected.toBuilder().apply {
-        metadataBuilder.timestamp = ts
-    }.build()
-    val assertActual = actual.toBuilder().apply {
-        metadataBuilder.timestamp = ts
-    }.build()
+
+    val assertExpected = when (expected.kindCase) {
+        AnyMessage.KindCase.MESSAGE -> {
+            expected.message.toBuilder().apply {
+                metadataBuilder.timestamp = ts
+            }.build()
+        }
+        AnyMessage.KindCase.RAW_MESSAGE -> {
+            expected.rawMessage.toBuilder().apply {
+                metadataBuilder.timestamp = ts
+            }.build()
+        }
+        else -> {
+            expected
+        }
+    }
+
+    val assertActual = when (actual.kindCase) {
+        AnyMessage.KindCase.MESSAGE -> {
+            actual.message.toBuilder().apply {
+                metadataBuilder.timestamp = ts
+            }.build()
+        }
+        AnyMessage.KindCase.RAW_MESSAGE -> {
+            actual.rawMessage.toBuilder().apply {
+                metadataBuilder.timestamp = ts
+            }.build()
+        }
+        else -> {
+            actual
+        }
+    }
+
     try {
         Assertions.assertEquals(assertExpected, assertActual, lazyMessage)
     } catch (e: AssertionFailedError) {
         throw AssertionFailedError(
-            "Error in message with type '${actual.messageType}'.\n${e.message}",
+            "Error in '${actual.kindCase}'.\n${e.message}",
             e.expected,
             e.actual,
             e.cause
         )
```

Comment:
Maybe create a method for rewrapping such exception? I've seen the same code somewhere above